src/librustc/infer/error_reporting/mod.rs

```diff
@@ -847,7 +847,7 @@ impl<'a, 'gcx, 'tcx> InferCtxt<'a, 'gcx, 'tcx> {
                                                "{} may not live long enough",
                                                labeled_user_string);
                 err.help(&format!("consider adding an explicit lifetime \
-                                   bound `{}: 'static`...",
+                                   bound `{}: 'static + `...",
```

Comment:
Hmm, this seems a bit odd to me. I guess you are imagining a case like this:

```rust
fn foo<T: Iterator>(t: &T) { bar(t); }

fn bar<T: 'static>(t: &T) { }

fn main() { }
```

which currently yields:

```
error[E0310]: the parameter type `T` may not live long enough
 --> src/main.rs:1:30
  |
1 | fn foo<T: Iterator>(t: &T) { bar(t); }
  |                              ^^^
  |
  = help: consider adding an explicit lifetime bound `T: 'static`...
note: ...so that the type `T` will meet its required lifetime bounds
 --> src/main.rs:1:30
  |
1 | fn foo<T: Iterator>(t: &T) { bar(t); }
  |                              ^^^
```

and your proposal would change to "consider adding an explicit lifetime bound `T: 'static +`...". But there are also cases where no `+` would be needed (e.g., just change the `T: Iterator` to `T` in the above example). And it seems weird to suggest a trailing `+` without something after it.

I wonder: if we just changed the term "bound" to "where-clause", would that be better? After all, one can always add `where T: 'static` in any case, without regard for whether `T: Iterator` is also declared elsewhere.

I wonder if we could use a "suggestion" or something instead. i.e., we have the ability to grab the declaration of `T`, and we could detect if there are existing bounds and then suggest `T: Iterator + 'static` -- i.e., show the existing bounds with `'static` in place. 

However, a quick glance suggests that this would take a fair bit of plumbing to achieve -- the only thing that's *readily* available is the name of the parameter and its index in the list. We'd have to add information about the type parameters in scope into the inference context or something like that.